scripts/create_baseline_stubs.py

```diff
@@ -90,13 +90,14 @@ async def get_upstream_repo_url(project: str) -> str | None:
 
         # Order the project URLs so that we put the ones
         # that are most likely to point to the source code first
-        urls_to_check: list[str] = []
         url_names_probably_pointing_to_source = ("Source", "Repository", "Homepage")
+
+        urls_to_check: list[str] = []
         for url_name in url_names_probably_pointing_to_source:
             if url := project_urls.get(url_name):
-                urls_to_check.append(url)
+                urls_to_check.append(url)  # noqa: PERF401 # False-positive with walrus, this code is the fastest form
         urls_to_check.extend(
-            url for url_name, url in project_urls.items() if url_name not in url_names_probably_pointing_to_source
+            [url for url_name, url in project_urls.items() if url_name not in url_names_probably_pointing_to_source]
         )
 
         for url in urls_to_check:
```

Comment:
Test:
```py
import timeit

setup = """
from itertools import chain

project_urls = {
    "Changelog": "https://code.qt.io/cgit/pyside/pyside-setup.git/tree/doc/changelogs",
    "Documentation": "https://doc.qt.io/qtforpython",
    "Homepage": "https://pyside.org",
    "Repository": "https://code.qt.io/cgit/pyside/pyside-setup.git/",
    "Tracker": "https://bugreports.qt.io/projects/PYSIDE",
}

url_names_probably_pointing_to_source = ("Source", "Repository", "Homepage")
"""

stmt = """
urls_to_check: list[str] = []
for url_name in url_names_probably_pointing_to_source:
    if url := project_urls.get(url_name):
        urls_to_check.append(url)
urls_to_check.extend(
    url for url_name, url in project_urls.items() if url_name not in url_names_probably_pointing_to_source
)

for url in urls_to_check:
    pass
"""
print(timeit.timeit(stmt=stmt, setup=setup))

stmt = """
urls_to_check: list[str] = []
for url_name in url_names_probably_pointing_to_source:
    if url := project_urls.get(url_name):
        urls_to_check.append(url)
urls_to_check.extend(
    [url for url_name, url in project_urls.items() if url_name not in url_names_probably_pointing_to_source]
)

for url in urls_to_check:
    pass
"""
print(timeit.timeit(stmt=stmt, setup=setup))

stmt = """
project_urls_probably_pointing_to_source = (
    project_urls.get(url_name) for url_name in url_names_probably_pointing_to_source
)
urls_to_check = chain(
    (url for url in project_urls_probably_pointing_to_source if url),
    (url for url_name, url in project_urls.items() if url_name not in url_names_probably_pointing_to_source),
)

for url in urls_to_check:
    pass
"""
print(timeit.timeit(stmt=stmt, setup=setup))

stmt = """
project_urls_probably_pointing_to_source = [project_urls.get(url_name) for url_name in url_names_probably_pointing_to_source]
urls_to_check = (
    [url for url in project_urls_probably_pointing_to_source if url]
    + [url for url_name, url in project_urls.items() if url_name not in url_names_probably_pointing_to_source]
)

for url in urls_to_check:
    pass
"""
print(timeit.timeit(stmt=stmt, setup=setup))
```

Results:
```py
0.6841839999979129
0.5984525999992911
0.9246598999998241
0.7370160999998916
```

Reply:
Nevermind, I was just missing parenthesis: https://github.com/astral-sh/ruff/issues/15047#issuecomment-2551977622
(new run time on python 3.13 is `0.5153441999973438`)